viewproduct.php

```diff
@@ -10,162 +10,157 @@
-               $single_product = mysqli_query($con, $query7);
-            
-            ///////////////////////Display reviews///////////////////////
-            
-            $select_reviews = "SELECT * FROM `tblproductreviews` WHERE `ProductID` LIKE '%".$PRODUCTID."%' ORDER BY ReviewDate ASC";
-            $result_reviews = mysqli_query($con, $select_reviews);
-            
-            ///////////////////////Display related products///////////////////////
-            
-            $select_category_of_product = mysqli_query($con, "SELECT CategoryID as categoryid FROM `tblproducts`  WHERE `ProductID`='".$PRODUCTID."' LIMIT 1");
-            $result_select_category_of_product = mysqli_fetch_assoc($select_category_of_product);
-            $CATEGORY_ID = $result_select_category_of_product["categoryid"];
-            
-            $select_random_products = "SELECT * FROM tblproducts WHERE CategoryID LIKE '%".$CATEGORY_ID."%'  ORDER BY RAND() LIMIT 4;";
-            $result_select_random_products = mysqli_query($con, $select_random_products);
-            
-            
-            }
+<?php 
+include('header.php');
+
+if(!isset($_GET["pid"])){
+    header("location: pnb.php");
+} else {
+    $PRODUCTID = $_GET["pid"];
+
+    ///////////////////////Display product///////////////////////
+
+    $query7 = "SELECT * FROM `tblproducts` WHERE `ProductID` = ? LIMIT 1";
+    $stmt_product = mysqli_prepare($con, $query7);
+    mysqli_stmt_bind_param($stmt_product, "s", $PRODUCTID);
+    mysqli_stmt_execute($stmt_product);
+    $single_product = mysqli_stmt_get_result($stmt_product);
+
+    ///////////////////////Display reviews///////////////////////
+
+    $select_reviews = "SELECT * FROM `tblproductreviews` WHERE `ProductID` = ? ORDER BY ReviewDate ASC";
+    $stmt_reviews = mysqli_prepare($con, $select_reviews);
+    mysqli_stmt_bind_param($stmt_reviews, "s", $PRODUCTID);
+    mysqli_stmt_execute($stmt_reviews);
+    $result_reviews = mysqli_stmt_get_result($stmt_reviews);
+
+    ///////////////////////Display related products///////////////////////
+
+    $select_category_of_product = "SELECT CategoryID as categoryid FROM `tblproducts` WHERE `ProductID` = ? LIMIT 1";
+    $stmt_category = mysqli_prepare($con, $select_category_of_product);
+    mysqli_stmt_bind_param($stmt_category, "s", $PRODUCTID);
+    mysqli_stmt_execute($stmt_category);
+    $result_select_category_of_product = mysqli_stmt_get_result($stmt_category);
+    $CATEGORY_ID = mysqli_fetch_assoc($result_select_category_of_product)["categoryid"];
+
+    $select_random_products = "SELECT * FROM tblproducts WHERE CategoryID = ? ORDER BY RAND() LIMIT 4";
+    $stmt_related = mysqli_prepare($con, $select_random_products);
+    mysqli_stmt_bind_param($stmt_related, "s", $CATEGORY_ID);
+    mysqli_stmt_execute($stmt_related);
+    $result_select_random_products = mysqli_stmt_get_result($stmt_related);
+}
```

Comment:
code could be improved by using parameterized queries instead of concatenating variables directly into the SQL 
statements to prevent SQL injection attacks. Here's an example of how to use prepared statements with mysqli:

if the queries were executed successfully before using their results. This can be done by checking the return value of 
mysqli_query() or mysqli_stmt_execute() and  handling any errors that may occur.